data/trie/sync.go

```diff
@@ -136,6 +136,8 @@ func (ts *trieSyncer) StartSyncing(rootHash []byte, ctx context.Context) error {
 	ts.rootFound = false
 	ts.rootHash = rootHash
 
+	defer ts.interceptedNodes.Clear()
```

Comment:
this should not be called here

Reply:
Refactored. This will be called after all the state has been synced.